src/services/ProcessWalletEvent.js

```diff
@@ -0,0 +1,106 @@
+import { WALLET_EVENTS } from '@/constants'
+
+export default class NetworkStatus {
+  constructor() {
+    this.currentState = null
+    this.isResyncButtonVisible = false
+    this.isLoadingVisible = false
+    this.isNodeSynced = true
+    this.isNodeDisconnected = false
+    this.isWalletSynced = false
+    this.syncError = false
+    this.progress = 0
+    this.timestamp = 0
+  }
+
+  currentStatus() {
+    if (this.isNodeDisconnected) {
+      this.currentState = {
+        label: 'NODE DISCONNECTED',
+        color: 'red',
```

Comment:
This shouldn't know anything about colors, we can create a label -> color map to get the color in the view.